adapters/p2p2mempool/transaction.go

```diff
@@ -42,9 +42,15 @@ func AdaptTransaction(t *mempooltransaction.MempoolTransaction, network *utils.N
 			return mempool.BroadcastedTransaction{}, err
 		}
 	case *mempooltransaction.MempoolTransaction_DeployAccountV3:
-		tx = p2p2core.AdaptDeployAccountV3TxnCommon(t.GetDeployAccountV3(), t.TransactionHash)
+		tx, err = p2p2core.AdaptDeployAccountV3TxnCommon(t.GetDeployAccountV3(), t.TransactionHash)
+		if err != nil {
+			return mempool.BroadcastedTransaction{}, err
+		}
 	case *mempooltransaction.MempoolTransaction_InvokeV3:
-		tx = p2p2core.AdaptInvokeV3TxnCommon(t.GetInvokeV3(), t.TransactionHash)
+		tx, err = p2p2core.AdaptInvokeV3TxnCommon(t.GetInvokeV3(), t.TransactionHash)
+		if err != nil {
+			return mempool.BroadcastedTransaction{}, err
+		}
```

Comment:
Since every case checks for an error now, I believe the error check can be written once outside the switch case statement